### PR TITLE
[SPARK-36511][MINOR][SQL] Remove ColumnIOUtil

### PR DESCRIPTION
**Pull request by panbingkun**
### What changes were proposed in this pull request?
This PR aim to remove redundant ColumnIOUtil.

### Why are the changes needed?
Complete todo.
image

reason: from parquet version 1.12.3, methods below are public
> 1.ColumnIO.getDefinitionLevel: https://github.com/apache/parquet-mr/blob/apache-parquet-1.12.3/parquet-column/src/main/java/org/apache/parquet/io/ColumnIO.java#L84-L86
> 2.ColumnIO.getRepetitionLevel: https://github.com/apache/parquet-mr/blob/apache-parquet-1.12.3/parquet-column/src/main/java/org/apache/parquet/io/ColumnIO.java#L77-L79
> 3.ColumnIO.getFieldPath: https://github.com/apache/parquet-mr/blob/apache-parquet-1.12.3/parquet-column/src/main/java/org/apache/parquet/io/ColumnIO.java#L50-L52

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
Pass GA.

### CODE REVIEW

**Review by HyukjinKwon — APPROVED** [review]

**Comment by srowen** [comment]
Merged to master

File: sql/core/src/main/java/org/apache/parquet/io/ColumnIOUtil.java [hunk 1]
```diff
@@ -1,40 +0,0 @@
...
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-
-package org.apache.parquet.io;
-
-/**
- * This is a workaround since methods below are not public in {@link ColumnIO}.
- *
- * TODO(SPARK-36511): we should remove this once PARQUET-2050 and PARQUET-2083 are released with
```
Comment (dongjoon-hyun) [inline sql/core/src/main/java/org/apache/parquet/io/ColumnIOUtil.java:23]: This is not a minor. This should be removed with JIRA issue, SPARK-36511 .

Comment (dongjoon-hyun) [reply sql/core/src/main/java/org/apache/parquet/io/ColumnIOUtil.java:23]: To @panbingkun , please respect the code when you remove.
Apache Spark community uses IDed TODO like this for trace-ability and we had better respect them.

**Comment by dongjoon-hyun** [comment]
Also, cc @sunchao since he is the issue reporter of SPARK-36511 .

**Comment by srowen** [comment]
OK fair enough can you file a JIRA? I can connect the dots

**Comment by dongjoon-hyun** [comment]
@srowen , the JIRA, SPARK-36511, was already filed and hard-coded in the removed part. Here, @panbingkun ignored it mistakenly.

**Comment by srowen** [comment]
Ah ok I'll connect it

**Comment by dongjoon-hyun** [comment]
Thank you, @srowen .

**Pull request closed without merging**